src/CheckIn.js

```diff
@@ -42,6 +42,7 @@ class CheckIn extends React.Component {
     handleSubmit: PropTypes.func.isRequired,
     pristine: PropTypes.bool,
     submithandler: PropTypes.func,
+    barcodeRef: PropTypes.object.isRequired,
```

Comment:
Could you please use more useful PropType rather than just object. I am thinking about `PropTypes.shape({ current: PropTypes.instanceOf(Element) })` for example. Check discussion [here](https://stackoverflow.com/questions/48007326/what-is-the-correct-proptype-for-a-ref-in-react) for referrence.